keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -188,6 +189,18 @@ def __init__(
         self.backbone = backbone
         self.preprocessor = preprocessor
 
+        # These layer attributes are used in cached decoding.
+        # We save them as class attrs to avoid excessive `get_layer` calls.
+        self.transformer_layers = []
```

Comment:
We should remove these lines here, and call `self.backbone.get_layer("xx")` directly in the methods that need them.

The issue with adding these layers to self is you are actually fundamentally changing the structure of our checkpoints! Keras will consider these new layers though the `setattr` tracking on the base layer.

Reply:
An alternative, if you are so inclined, is to move those setattr calls to a new method (`set_layer_attributes()`?) decorated with a do-not-track decorator. Though the `@property` accessors are fine too.